command/agent/config/config_test.go

```diff
@@ -424,63 +631,98 @@ func TestLoadConfigFile_AgentCache_NoAutoAuth(t *testing.T) {
+		t.Fatal("ValidateConfig should return an error when use_auto_auth_token=true and no auto_auth section present")
 	}
 }
 
 func TestLoadConfigFile_Bad_AgentCache_ForceAutoAuthNoMethod(t *testing.T) {
-	_, err := LoadConfig("./test-fixtures/bad-config-cache-force-auto_auth.hcl")
+	config, err := LoadConfigFile("./test-fixtures/bad-config-cache-force-token-no-auth-method.hcl")
+	if err != nil {
+		t.Fatalf("LoadConfigFile should not return an error for this config, err: %v", err)
+	}
+	if config == nil {
+		t.Fatal("config was nll")
+	}
+	err = config.ValidateConfig()
 	if err == nil {
-		t.Fatal("LoadConfig should return an error when use_auto_auth_token=force and no auto_auth section present")
+		t.Fatal("ValidateConfig should return an error when use_auto_auth_token=force and no auto_auth section present")
 	}
 }
 
 func TestLoadConfigFile_Bad_AgentCache_NoListeners(t *testing.T) {
-	_, err := LoadConfig("./test-fixtures/bad-config-cache-no-listeners.hcl")
-	if err == nil {
-		t.Fatal("LoadConfig should return an error when cache section present and no listeners present and no templates defined")
+	_, err := LoadConfigFile("./test-fixtures/bad-config-cache-no-listeners.hcl")
+	if err != nil {
+		t.Fatalf("LoadConfigFile should return an error for this config")
 	}
 }
 
 func TestLoadConfigFile_Bad_AutoAuth_Wrapped_Multiple_Sinks(t *testing.T) {
-	_, err := LoadConfig("./test-fixtures/bad-config-auto_auth-wrapped-multiple-sinks.hcl")
+	_, err := LoadConfigFile("./test-fixtures/bad-config-auto_auth-wrapped-multiple-sinks.hcl")
 	if err == nil {
-		t.Fatal("LoadConfig should return an error when auth_auth.method.wrap_ttl nonzero and multiple sinks defined")
+		t.Fatalf("LoadConfigFile should return an error for this config, err: %v", err)
 	}
 }
 
 func TestLoadConfigFile_Bad_AutoAuth_Nosinks_Nocache_Notemplates(t *testing.T) {
-	_, err := LoadConfig("./test-fixtures/bad-config-auto_auth-nosinks-nocache-notemplates.hcl")
+	config, err := LoadConfigFile("./test-fixtures/bad-config-auto_auth-nosinks-nocache-notemplates.hcl")
+	if err != nil {
+		t.Fatalf("LoadConfigFile should not return an error for this config, err: %v", err)
+	}
+	if config == nil {
+		t.Fatal("config was nll")
+	}
+	err = config.ValidateConfig()
 	if err == nil {
-		t.Fatal("LoadConfig should return an error when auto_auth configured and there are no sinks, caches or templates")
+		t.Fatal("ValidateConfig should return an error when auto_auth configured and there are no sinks, caches or templates")
 	}
 }
 
 func TestLoadConfigFile_Bad_AutoAuth_Both_Wrapping_Types(t *testing.T) {
-	_, err := LoadConfig("./test-fixtures/bad-config-method-wrapping-and-sink-wrapping.hcl")
+	_, err := LoadConfigFile("./test-fixtures/bad-config-method-wrapping-and-sink-wrapping.hcl")
 	if err == nil {
-		t.Fatal("LoadConfig should return an error when auth_auth.method.wrap_ttl nonzero and sinks.wrap_ttl nonzero")
+		t.Fatalf("LoadConfigFile should not return an error for this config")
```

Comment:
Your failure message now says "should not return an error", but the condition is the opposite (err is nil).

Reply:
Good catch! Thanks